src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -2383,9 +2383,7 @@ void MacroAssembler::verify_secondary_supers_table(Register r_sub_klass,
   addi(r_array_base, r_array_base, Array<Klass*>::base_offset_in_bytes());
 
   // convert !=0 to 1
-  neg(R0, result);
-  orr(result, result, R0);
-  srdi(result, result, 63);
+  normalize_bool(result, true);
```

Comment:
wait, is build successful with this code change ?

Reply:
I think you can't specify the parameters arbitrarily like this. If you want to pass "true" then you have to specify the register as well;

Reply:
Build is succeding . And i also did a register check for templateGenerator with this example using normalize_bool(result,false) ,(java -version was not hitting this code). Seems to be working fine.

Reply:
I think the 2nd value is taken at default.

Reply:
```cpp
amitkumar@Amits-MacBook-Pro ~ % cat tt.cpp
#include <iostream>
using namespace std; 

void fun(int a, char b = 'a', float c = 10.3); 

int main(void) {
  fun(10); 
  fun(10, 103.483);
  return 0;
}

void fun(int a, char b, float c) {
}

amitkumar@Amits-MacBook-Pro ~ % g++ tt.cpp 
tt.cpp:8:11: warning: implicit conversion from 'double' to 'char' changes value from 103.483 to 103 [-Wliteral-conversion]
  fun(10, 103.483);
  ~~~     ^~~~~~~
1 warning generated.
```

That's weird!!!

Reply:
Yeah i am not sure why the warning did not pop up during build.It run through fine. Thanks for pointing it out. I will correct it.

Reply:
Maybe this needs to be enforced during build then ? @offamitkumar

Reply:
> Maybe this needs to be enforced during build then ?

In this type of conversion we get a warning, And hotspot code treats all of the warnings as error. So this should break the build. 

But what I suspect is that as registers are being treated as integers, we can do this `Register temp = 1`. This is valid syntax.  So with `normalise_bool(result, true)` you set `Register dst = result`, `Register temp=1(true)` and `is_64bit` was considered `default`. Seems like there is nothing wrong syntax-wise. Hence no warnings were generated.